src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -787,8 +787,8 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof PasswordCallback) {
                     final PasswordCallback passwordCallback = (PasswordCallback) callback;
-
-                    if (getCredentialAcquireSupport(PasswordCredential.class).mayBeSupported()) {
+                    SupportLevel supportLevel = getCredentialAcquireSupport(PasswordCredential.class);
```

Comment:
Right I agree.  I think static analysis may be returning a possible null here because there might be, say, one incorrectly implemented security realm.  I have some thoughts on this I'd like to share in chat, which I will do right now!